Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -0,0 +1,196 @@
+  | _ => return Array.mk [(StyleError.copyright none, 0)]
+
+  -- If this is given, we go over the individual lines one by one,
+  -- and provide some context on what is mis-formatted (if anything).
+  let mut output := Array.mkEmpty 0
+  -- By hypotheses above, start has at least five lines, so the `none` cases below are never hit.
+  -- The first real line should state the copyright.
+  if let some copy := start.get? 1 then
+    if !(copy.startsWith "Copyright (c) 20" && copy.endsWith ". All rights reserved.") then
+      output := output.push (StyleError.copyright "Copyright line is malformed", 2)
+  -- The second line should be standard.
+  let expected_second_line := "Released under Apache 2.0 license as described in the file LICENSE."
+  if start.get? 2 != some expected_second_line then
+    output := output.push (StyleError.copyright
+      s!"Second line should be \"{expected_second_line}\"", 3)
+  -- The third line should contain authors.
+  if let some line := start.get? 3 then
+    if !line.containsSubstr "Author" then
+      output := output.push (StyleError.copyright
+        "The third line should describe the file's main authors", 4)
+    else
+      -- If it does, we check the authors line is formatted correctly.
+      if !is_correct_authors_line line then
+        output := output.push (StyleError.authors, 4)
+  return output
+
+
+/-- Whether a collection of lines consists *only* of imports:
+in practice, this means it's an imports-only file and exempt from almost all linting. -/
+def is_imports_only_file (lines : Array String) : Bool :=
+  -- The Python version also excluded comments: since the import-only files are
+  -- automatically generated and don't contains comments, this is in fact not necessary.
+  -- XXX: also implement parsing of multi-line comments.
+  lines.all (fun line ↦ line.startsWith "import " || line == "" || line.startsWith "-- ")
+
+end
+
+/-- All text-based linters registered in this file. -/
+def all_linters : Array LinterCore := Array.mk
+  [copyright_header]
+
+/-- Read a file, apply all text-based linters and return the formatted errors.
+
+Return `true` if there were new errors (and `false` otherwise). -/
+def lint_file (path : System.FilePath) : IO Bool := do
+  let lines ← IO.FS.lines path
+  -- We don't need to run any checks on imports-only files.
+  -- NB. The Python script used to still run a few linters; this is in fact not necessary.
+  if is_imports_only_file lines then
+    return false
+  let all_output := (Array.map (fun lint ↦
+    (Array.map (fun (e, n) ↦ ErrorContext.mk e n path)) (lint lines))) all_linters
+  -- XXX: this list is currently not sorted: for github, that's probably fine
+  let errors := Array.flatten all_output
+  formatErrors errors
+  return errors.size > 0
+
+/-- Lint all files referenced in a given import-only file.
+Return the number of files which had new style errors. -/
+def lint_all_files (path : System.FilePath) : IO UInt32 := do
```

Comment:
nit: this, and most of the other names in the file, do not match the usual naming conventions.

Reply:
Indeed, you're right. Addressed.